### PR TITLE
Public sharing improvements

### LINKED ISSUE #232: Optically highlight which link is selected
georg cloud share

The current selection (either CalDAV or WebCal) should be highlighted just like the current view.

### LINKED ISSUE #240: Show proper emptycontent view for non-existing public calendar links
Currently when you go to https://cloud.nextcloud.com/apps/calendar/public/non-existing-calendar there’s just an empty calendar. This calendar does not actually exist and there should be indication of that. Instead of the calendar view on the right, there should be an emptycontent-like view similar to the Files app for empty folders. :)

Saying something like:

> [icon-calendar]
> **Calendar does not exist**
> Maybe the link is wrong?

cc @nextcloud/calendar

### LINKED ISSUE #243: Calendar shared by link: overall design & left navigation
Currently a calendar shared by link looks like this:
capture du 2016-12-15 02-36-40

Some thoughts:
- There’s no blue bar on top which clearly states »Nextcloud«. It’s not only a branding thing but also to know where you are. We do that everywhere else when sharing – Files, Spreed, Gallery etc.
- The left navigation looks a bit lost. We could have the calendar name a bit bigger like a h2 maybe? Similar to the »Files drop« feature in the server
- Also similar to the »Files drop«, it would be cool to not only show the user id in parentheses like `(jan)` but actually show a separate line with avatar and full name (fallback to username)
- Bug: the right border of the left navigation is not going through the whole way. The lower half (except next to settings) is basically missing.

What do you think @nextcloud/calendar @nextcloud/designers?

### LINKED ISSUE #248: Mobile menu position incorrect
Seems related to calendar in public mode only.
screenshot_20161215-133931
screenshot_20161215-133939

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #232
fixes #240
fixes #243 
fixes #248

### CODE REVIEW

**Comment by georgehrke** [comment]
requires https://github.com/nextcloud/calendar/pull/373 for the avatar directive

**Comment by georgehrke** [comment]
Just added the empty content warning:
nextcloud chromium today at 4 52 47 pm

**Comment by georgehrke** [comment]
I would suggest the following changes
- I moved download and subscribe (iCal Feed) to the upper right corner. I don't think we actually need the CalDAV link. Everywhere on the web where you can subscribe to a calendar it's webcal. I've never seen CalDAV before. It's probably more confusing than beneficial.
- If we decide to remove the CalDAV link, rename "Subscribe & embed" to "Embed and change the icon to something like "</>"
- Hide the embed section completely when viewing the calendar that is embedded itself.

nextcloud chromium today at 5 04 21 pm

(Additionally, I will add the user avatar above the calendar name)

**Comment by tcitworld** [comment]
> Maybe you got the wrong link

Or the calendar got unshared.

**Comment by tcitworld** [comment]
> Maybe remove Embed from the public sharing site completely and show it inside the calendar app

Why not, but I guess we already have issues to place things in the calendar app itself, so...

I agree with the rest. Nice work !

**Comment by georgehrke** [comment]
i changed the way we generate the public sharing url.
Reminder to myself:
- [ ] test with a nextcloud instance in a subfolder
- [ ] test with calendar app in a second appfolder

**Comment by georgehrke** [comment]
The only thing missing here is the avatar.
I need the avatar directive from #373 for that :)

**Comment by tcitworld** [comment]
So let's merge #373 for now and then open another PR for global work on the sidebar ?

**Comment by georgehrke** [comment]
Hm. Yes. I think that would be the best solution :)

**Comment by georgehrke** [comment]
New navigation bar of public sharing site:
nextcloud chromium today at 12 04 45 pm

**Comment by tcitworld** [comment]
Oh. Maybe put ~~your face~~ the avatar a bit smaller ? :)

**Comment by georgehrke** [comment]
Hm. yes.

It's 128px, just like the public file sharing, at the moment, but there is also much more whitespace on the public file sharing site.
Resizing it down to 96 ...

File: templates/part.publicinformations.php [hunk 1]
```diff
@@ -22,51 +22,30 @@
...
-	<span class="action permanent displayname">{{ item.calendar.displayname }}</span>
-
-	<span class="icon-download svg public-ics-download"
-		ng-click="download(item)">
-		<?php p($l->t('Download')); ?>
-	</span>
+<div id="scollable" class="settings-fieldset-interior public-left-side" ng-repeat="item in calendarListItems">
+	<div class="avatardiv" data-user="{{ item.calendar.owner }}" data-size="96" avatar></div>
+	<h3 class="action permanent displayname" ngCloak>
+		<?php print_unescaped($l->t('%s shared the calendar <strong>%s</strong> with you', ['{{ item.getOwnerName() }}', '{{ item.getPublicDisplayname() }}'])); ?>
```
Comment (georgehrke) [inline templates/part.publicinformations.php]: @LukasReschke Is this safe?

**Comment by georgehrke** [comment]
Jan had issues with Unknown timezone, caused by changed $rootScope.baseUrl

**Comment by jancborchardt** [comment]
Also, clicking the link icon in the sharing section opens a new tab of the files app instead of the shared calendar.

And instead of the calendar favicon, for the emptycontent please use `.icon-calendar-dark` :)

**Comment by georgehrke** [comment]
@jancborchardt The unknown timezone issue should be gone now

**Comment by georgehrke** [comment]
> Also, clicking the link icon in the sharing section opens a new tab of the files app instead of the shared calendar.

This was caused by the same issue that caused the "Unknown timezone" thing.
Should be fixed now as well

**Comment by georgehrke** [comment]
please review @tcitworld @raimund-schluessler @jancborchardt @eppfel :)

File: templates/main.php [hunk 2]
```diff
@@ -56,6 +56,16 @@
...
+<style>
+	@media only screen and (max-width: 768px) {
+		#app-navigation, #app-content {
+			top: 45px !important;
+		}
+	}
+</style>
+<?php endif; ?>
+
 <?php if($_['isEmbedded']): ?>
```
Comment (tcitworld) [inline templates/main.php]: Maybe add $_['isPublic'] here too.

Comment (georgehrke) [inline templates/main.php]: isEmbedded will only ever be treu when isPublic is true too

**Comment by georgehrke** [comment]
nextcloud chromium today at 4 39 30 pm

That's the new 404 screen with the grey calendar icon

**Comment by georgehrke** [comment]
please test this branch again to make sure the timezone error is gone :)

**Comment by jancborchardt** [comment]
Great, the timezone issue is fixed. :)

Check please review too @nextcloud/calendar

**Comment by georgehrke** [comment]
@jancborchardt Fixed the flashing text issue with my latest commit :)

File: templates/main.php (hunk 2 above)
Comment (georgehrke) [inline templates/main.php]: @tcitworld Are you fine with not adding isPublic here?

**Comment by tcitworld** [comment]
Maybe it's related to changes in core, but the users pic in the sidebar resolves for me to `/index.php/avatar//96`, which is of course not found.

**Comment by georgehrke** [comment]
The data-user attr is empty. I don't think it's a core issue, but rather related to the latest ng update

**Comment by georgehrke** [comment]
@tcitworld please pull latest master and check again :)

**Comment by jancborchardt** [comment]
Ok, so is this good to go? :) @raghunayyar @tcitworld did you review? :)

**Comment by georgehrke** [comment]
We will have to make use of https://github.com/nextcloud/server/pull/4401 for master and Nextcloud 12, but I would like to address that in a follow-up PR.

**Comment by georgehrke** [comment]
ping @tcitworld :)

**Comment by georgehrke** [comment]
I will go ahead and merge this tomorrow evening if no one objects

**Pull request merged**